### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by ZainNoor99**

### CODE REVIEW

**Comment by jhorbulyk** [comment]
Solves #7

File: README.md [hunk 1]
```diff
@@ -1,50 +1,37 @@
-
 # xml-component
 
 ## Description
-This is an open source component for Converting XML to and from JSON on elastic.io platform.
-Component to be used on the elastic.io platform, which is able to convert XML to and from JSON. 
-### Purpose
-This component has 3 actions allowing users to pass in either generic but well format XML/JSON string or XML attachment and produces a generic string of the other file type. The output then can be maped and used in other components. 
+Component to convert file types.
```
Comment (jhorbulyk) [inline README.md]: `to convert file types` Change to `to convert between xml and json data`

File: README.md [hunk 2]
```diff
@@ -1,50 +1,37 @@
...
-### How it works. 
-Before you can deploy any code into elastic.io you must be a registered elastic.io platform user. Please see our home page at http://www.elastic.io to learn how.
-
-### Requirements
-Provided XML document (for `XML to JSON`) should be [well-formed](https://en.wikipedia.org/wiki/Well-formed_document) in order to be parsed correctly.
-You will get an error otherwise. 
-#### Environment variables 
-No environment variables need to be set.
+### Purpose
+Allows users to convert attachments to JSON or XML to and from JSON
```
Comment (jhorbulyk) [inline README.md]: It allows the converstion of both XML attachments and strings to & from JSON.

File: README.md [hunk 3]
```diff
@@ -1,50 +1,37 @@
...
-This component has 3 actions allowing users to pass in either generic but well format XML/JSON string or XML attachment and produces a generic string of the other file type. The output then can be maped and used in other components. 
+Component to convert file types. 
 
-### How it works. 
-Before you can deploy any code into elastic.io you must be a registered elastic.io platform user. Please see our home page at http://www.elastic.io to learn how.
-
-### Requirements
-Provided XML document (for `XML to JSON`) should be [well-formed](https://en.wikipedia.org/wiki/Well-formed_document) in order to be parsed correctly.
-You will get an error otherwise. 
-#### Environment variables
```
Comment (jhorbulyk) [inline README.md]: Leave this in please.

File: README.md [hunk 4]
```diff
@@ -1,50 +1,37 @@
...
+[well-formed](https://en.wikipedia.org/wiki/Well-formed_document) in order to be parsed correctly. 
+You will get an error otherwise. 
 
-#### Schemas 
-[input schema](lib/schemas/xmlToJson.in.json) \
-[output schema](lib/schemas/xmlToJson.out.json)
+### XML Attachment to JSON
+Looks at the JSON array of attachments passed in to component and converts all XML that it finds to generic JSON objects
+As input, the user can enter a patter pattern for filtering files by name or leave this field empty for processing all 
+incoming *.xml files.
```
Comment (jhorbulyk) [inline README.md]: Can you mention that there is one outbound message per matching attachment?

**Comment by ZainNoor99** [comment]
Done

File: README.md [hunk 5]
```diff
@@ -1,18 +1,17 @@
-
-# xml-component
+# XML Component
 
 ## Description
-This is an open source component for Converting XML to and from JSON on elastic.io platform.
-Component to be used on the elastic.io platform, which is able to convert XML to and from JSON. 
-### Purpose
-This component has 3 actions allowing users to pass in either generic but well format XML/JSON string or XML attachment and produces a generic string of the other file type. The output then can be maped and used in other components. 
+Component to convert between XML and JSON data.
```
Comment (stas-fomenko) [inline README.md]: elastic.io iPaas component to convert between XML and JSON data.

File: README.md [hunk 6]
```diff
@@ -21,30 +20,23 @@ No environment variables need to be set.
...
- - Action does not support local agents due to current platform limitations.
+Treats incoming message body as JSON and converts it to a generic XML string.
+
+## Known limitations
+ - The maximum size of incoming file for processing is 5 MiB. If the size of incoming file will be more than 5 MiB, 
+ action will throw error `Attachment *.xml is to large to be processed by XML component. File limit is: 5242880 byte, 
+ file given was: * byte.`. 
+ - `XML Attachemnt to JSON` action does not support local agents due to current platform limitations.
+ 
+ ##Additional Info
```
Comment (stas-fomenko) [inline README.md]: ## Additional Info

**Pull request merged**